### PR TITLE
feat: support gorm.Model auto increment keys using sequences

### PR DESCRIPTION
**Pull request by rahul2393**

### CODE REVIEW

**Comment by rahul2393** [comment]
@olavloite can you please help review

File: migrator_test.go [hunk 1]
```diff
@@ -144,52 +151,3 @@ func setupMockedTestServerWithConfigAndClientOptions(t *testing.T, config spanne
 		serverTeardown()
 	}
 }
-
```
Comment (olavloite) [inline migrator_test.go:147]: nit: everything from here to the end of the file seems unrelated. Could this be moved to a separate PR?

File: migrator.go [hunk 2]
```diff
@@ -108,7 +96,18 @@ func (m spannerMigrator) CreateTable(values ...interface{}) error {
...
-
+			for _, f := range stmt.Schema.Fields {
+				// Cloud spanner does not support auto incrementing primary keys.
+				if f.AutoIncrement && f.HasDefaultValue && f.DefaultValue == "" && f.DefaultValueInterface == nil {
+					if err := tx.Exec("CREATE SEQUENCE IF NOT EXISTS " +
+						stmt.Table + "_GORM_SEQUENCE" +
+						` OPTIONS (sequence_kind = "bit_reversed_positive")`).Error; err != nil {
+						return err
+					}
+					f.HasDefaultValue = true
```
Comment (olavloite) [inline migrator.go]: Based on the above `if` statement, this should already be true, and thus redundant.

File: migrator.go [hunk 3]
```diff
@@ -45,25 +44,15 @@ func (m spannerMigrator) AutoMigrate(values ...interface{}) error {
...
-	}
-	result := reflect.ValueOf(&m.Migrator).MethodByName("AutoMigrate").Call(inputs)
-	if len(result) != 1 {
-		return fmt.Errorf("unexpected return value length: %d", len(result))
-	}
-	if err, ok := result[0].Interface().(error); ok {
-		return err
-	}
-	if result[0].IsNil() {
+	err := m.Migrator.AutoMigrate(values...)
```
Comment (olavloite) [inline migrator.go:51]: I'm surprised that this works. I tried this multiple times as well, as it is by far the most logical way, but I was never able to get it work (and I don't remember why....)

File: migrator.go [hunk 4]
```diff
@@ -108,7 +96,18 @@ func (m spannerMigrator) CreateTable(values ...interface{}) error {
 				values                  = []interface{}{m.CurrentTable(stmt)}
 				hasPrimaryKeyInDataType bool
 			)
-
+			for _, f := range stmt.Schema.Fields {
+				// Cloud spanner does not support auto incrementing primary keys.
+				if f.AutoIncrement && f.HasDefaultValue && f.DefaultValue == "" && f.DefaultValueInterface == nil {
+					if err := tx.Exec("CREATE SEQUENCE IF NOT EXISTS " +
+						stmt.Table + "_GORM_SEQUENCE" +
```
Comment (olavloite) [inline migrator.go]: What is the max table name length and max sequence name length? Put another way: would it make sense to make the sequence name as 'short' as possible, by using something like `TableName + "_seq"`, in order to reduce the probability that a long table name will cause the sequence creation to fail because the name is too long?

Also; Should we add support for an annotation to set a custom sequence name? That won't work with `gorm.Model`, but would work with anyone who creates their own model and wants to use a custom sequence name.

Comment (rahul2393) [reply migrator.go]: Added support for defining tag "gorm_sequence_name" and tests for the same
